fe/lib/components/BrandSelect/BrandSelect.tsx

```diff
@@ -66,6 +72,26 @@ export const BrandSelect = ({
 
   const [selectedBrandId, setSelectedBrandId] = useState(initialBrandId);
 
+  useEffect(() => {
+    if (
+      hasPreselected &&
+      selectedBrandId &&
+      previousVariables?.hirerId !== hirerId
+    ) {
+      // Clear stale state of the previous hirer when `hirerId` changes.
+      onSelect?.();
+
+      setSelectedBrandId(initialBrandId);
```

Comment:
I'm also wondering whether we should validate that this `initialBrandId` belongs to the specified hirer 🤔